src/Microsoft.Health.Api/Features/Audit/AuditEgressMiddleware.cs

```diff
@@ -43,14 +43,7 @@ public async Task Invoke(HttpContext context)
             finally
             {
                 var statusCode = (HttpStatusCode)context.Response.StatusCode;
-
-                // Since authorization filters runs first before any other filters, if the authorization fails,
-                // the AuditLoggingFilterAttribute, which is where the audit logging would normally happen, will not be executed.
-                // This middleware will log any Unauthorized request if it hasn't been logged yet.
-                if (statusCode == HttpStatusCode.Unauthorized)
-                {
-                    _auditHelper.LogExecuted(context, _claimsExtractor);
-                }
+                _auditHelper.LogExecuted(context, _claimsExtractor);
```

Comment:
I'm not sure what the net effect is, but in thinking through it, we may have mismatched AuditExecuted events. Have you testing what happens with the custom endpoints like `/health/check` and [SchemaController](https://github.com/microsoft/healthcare-shared-components/blob/master/src/Microsoft.Health.SqlServer.Api/Controllers/SchemaController.cs)?

Reply:
What is meant by mismatched AuditExecuted events?
I did test for /health/check and it is working as before i.e. not getting logged. According to AuditHelper (in FHIR and DICOM), audit entries are only mapped for actions with assigned Audit Event Type. Since /health/check does not have an assigned Audit Event Type, it is not logged. Moving LogExecuted to AuditEgressMiddleware has not changed the existing functionality. But I would like to test specific scenarios if you could guide me through them.

Reply:
Sorry, I meant the calls to IAuditHelper  => LogExecuting / LogExecuted. Ok cool, just wanted to ensure we covered these utility endpoints.

Reply:
Yep, I made sure those are getting called as before :)

Reply:
Actually I did not know about the concept of Batch in FHIR. This won't work because of how batch requests are handled. This implementation would end up logging executed just once for the entire batch instead of logging for each sub-request within the batch. I will close this PR and put something together such that FHIR is not affected.